dap_job_quality/pipeline/eyp/ngram_analysis/get_ngrams_and_matches.py

```diff
@@ -0,0 +1,181 @@
+from datetime import datetime
+from nltk import ngrams
+import numpy as np
+import pandas as pd
+import re
+from sklearn.metrics.pairwise import cosine_similarity
+from sentence_transformers import SentenceTransformer
+from typing import List, Tuple, Union
+
+from dap_job_quality import logging, BUCKET_NAME
+from dap_job_quality.getters.afs_data import get_jq_sentences
+from dap_job_quality.getters.data_getters import save_to_s3
+from dap_job_quality.getters.keywords import get_keywords
+
+sent_model = SentenceTransformer("all-MiniLM-L6-v2")
+
+today = datetime.today().strftime("%Y-%m-%d")
+
+
+def split_ngrams(
+    text: str, length: int = 6, n: int = 4
+) -> Union[List[Tuple[str, ...]], List[str]]:
+    """
+    Split the input text into n-grams if the number of words in the text exceeds the specified length.
+
+    Args:
+        text (str): The input text to be split into n-grams.
+        length (int): The maximum length of text in words before splitting into n-grams. Default is 6.
+        n (int): The number of words in each n-gram. Default is 4.
+
+    Returns:
+        Union[List[Tuple[str, ...]], List[str]]: A list of n-grams (each n-gram is a tuple of words) if the text
+                                                 exceeds the specified length; otherwise, a list containing the
+                                                 original text.
+    """
+    if len(text.split()) > length:
+        ngram_list = list(ngrams(text.split(), n))
```

Comment:
Probably a minor speed gain, but this might help:
```suggestion
    text_words = text.split()
    if len(text_words) > length:
        ngram_list = list(ngrams(text_words, n))
```